### PR TITLE
[DEV-3777] Creates API for interactions

### PR DESCRIPTION
**Pull request by cdanwards**
### What does this PR do, or why is it needed?

Creates the API work needed to create interactions in Rock when someone prays.

### How do I test this PR?

1) Yarn Start
2) Go to your localhost:4000 and run a `incrementPrayed` on a prayer like `PrayerRequest:ab914f6fe3271e951cde1ef1deb1736d`.
3) Go to https://alpha-rock.newspring.cc/page/147 and run:
```
SELECT
    TOP 10 *
FROM
    [Interaction]
ORDER BY
    InteractionDateTime DESC
```
Then look to see if your interaction shows up.

---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [DEV-XXX](#url)
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android if applicable
- [ ] Set two relevant reviewers

## REVIEW

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Review by IsaacHardy — CHANGES_REQUESTED** [review]
Make sure you run your test and retrieve the new snapshots

**Comment by cdanwards** [comment]
@IsaacHardy The snapshots didn't change when I updated the tests.

File: packages/apollos-data-prayer/src/prayer-requests/schema.js [hunk 1]
```diff
@@ -22,6 +22,11 @@ const prayerRequestSchema = gql`
     flagRequest(id: String!): PrayerRequest
     savePrayer(nodeId: String!): PrayerRequest
     unSavePrayer(nodeId: String!): PrayerRequest
+    createPrayerInteraction(
+      prayerId: String!
+      operationName: String!
+      prayerTitle: String
```
Comment (IsaacHardy) [inline packages/apollos-data-prayer/src/prayer-requests/schema.js]: This probably needs to be removed.

**Review by IsaacHardy — CHANGES_REQUESTED** [review]

**Comment by cdanwards** [comment]
We're not testing any of the other data-sources in our custom api, so I didn't write the tests for those.

File: packages/apollos-data-prayer/src/prayer-requests/schema.js (hunk 1 above)
Comment (cdanwards) [reply packages/apollos-data-prayer/src/prayer-requests/schema.js]: Removed.

**Comment by IsaacHardy** [comment]
@cdanwards We test data-sources only when it's appropriate. Since we've edited a DS and it's not being tested by calling the resolver, we'll need to test the DS method itself.

File: packages/apollos-church-api/config.yml [hunk 2]
```diff
@@ -54,8 +54,9 @@ ROCK_MAPPINGS:
   INTERACTIONS:
     # Name of the InteractionChannel that will be created
     CHANNEL_NAME: Apollos NewSpring App
-    # Name of the InteractionComponent that will be created for Content Items
+    # Name of the InteractionComponent that will be created for Content Items or Prayer Requests
     COMPONENT_NAME: Apollos NewSpring Content Item
+    COMPONENT_NAME_PRAYER_REQUEST: Apollos NewSpring Prayer Request
```
Comment (IsaacHardy) [inline packages/apollos-church-api/config.yml]:
```suggestion
    PRAYER_REQUEST: Apollos NewSpring Prayer Request
```

File: packages/apollos-church-api/src/data/rock-constants/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      } - ${prayerName || prayerRequestId}`,
+      channelId: channel.id,
+      entityId: parseInt(prayerRequestId, 10),
+    });
+  }
+
+  async prayerRequestInteractionChannel() {
+    const { id } = await this.prayerModelType('PrayerRequest');
+    return this.createOrFindInteractionChannel({
+      channelName: ROCK_MAPPINGS.INTERACTIONS.CHANNEL_NAME,
```
Comment (IsaacHardy) [inline packages/apollos-church-api/src/data/rock-constants/index.js:24]: Why wouldn't this be `COMPONENT_NAME_PRAYER_REQUEST` as well?

File: packages/apollos-church-api/src/data/rock-constants/index.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { RockConstants as apollosRockConstants } from '@apollosproject/data-connector-rock';
+import ApollosConfig from '@apollosproject/config';
+
+const { ROCK_MAPPINGS } = ApollosConfig;
+
+const mapNewSpringNameToRockName = (name) => {
```
Comment (IsaacHardy) [inline packages/apollos-church-api/src/data/rock-constants/index.js]: Why is this function even needed? All it does is say if 'PrayerRequest' equals 'PrayerRequest'... Which it always would.

File: packages/apollos-church-api/src/data/rock-constants/index.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  async prayerRequestInteractionChannel() {
+    const { id } = await this.prayerModelType('PrayerRequest');
+    return this.createOrFindInteractionChannel({
+      channelName: ROCK_MAPPINGS.INTERACTIONS.CHANNEL_NAME,
+      entityTypeId: id,
+    });
+  }
+
+  async prayerModelType(nameInput) {
+    const name = mapNewSpringNameToRockName(nameInput);
```
Comment (IsaacHardy) [inline packages/apollos-church-api/src/data/rock-constants/index.js]: I think this call is not needed.

File: packages/apollos-church-api/src/data/rock-constants/index.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    const types = await this.request('EntityTypes')
+      .filter(`Name eq 'Rock.Model.${name}'`)
```
Comment (IsaacHardy) [inline packages/apollos-church-api/src/data/rock-constants/index.js]: just insert 'PrayerRequest' for name.

File: packages/apollos-church-api/src/data/rock-constants/index.js (hunk 3 above)
Comment (cdanwards) [reply packages/apollos-church-api/src/data/rock-constants/index.js:24]: This indicates where the interaction is coming from, `NewSpring Apollos App`

Comment (IsaacHardy) [reply packages/apollos-church-api/src/data/rock-constants/index.js:24]: Oh. I read `COMPONENT_NAME` for some reason.

File: packages/apollos-church-api/config.yml (hunk 2 above)
Comment (cdanwards) [reply packages/apollos-church-api/config.yml]: Resolved.

File: packages/apollos-church-api/src/data/rock-constants/index.js (hunk 4 above)
Comment (cdanwards) [reply packages/apollos-church-api/src/data/rock-constants/index.js]: Resolved.

**Review by IsaacHardy — DISMISSED** [review]
LGTM

**Review by IsaacHardy — APPROVED** [review]
LGTM

**Pull request merged**